packages/server-core/src/user/identity-provider/identity-provider.hooks.ts

```diff
@@ -185,9 +187,15 @@ async function addIdentityProviderType(context: HookContext<IdentityProviderServ
   if (
     !isAdmin &&
     context.params!.provider &&
-    !['password', 'email', 'sms'].includes((context!.actualData as IdentityProviderData).type)
+    !['password', 'email', 'sms'].includes((context!.actualData as IdentityProviderData).type as string)
   ) {
-    ;(context.data as IdentityProviderData).type = 'guest'
+    if (!USER_ID_REGEX.test((context.data as IdentityProviderData).token as string))
```

Comment:
It would be nice to rip out the type casts all over the place and just have `const data = context.data as IdentityProviderData` that is then used to make it easier to read.